### ISSUE TITLE
Deal with empty content nodes

### ISSUE THREAD

**Issue by christianp** [mathjax/ctop#29]
http://www.w3.org/Math/testsuite/build/main/ErrorHandling/NumChildren/noChildContent-form.xhtml

These should put some kind of error tag in.

**Comment by pkra** [comment]
I think this test reveals a regular MathJax bug with empty annotation tags.

**Comment by davidcarlisle** [comment]
On 2 May 2014 10:59, Peter Krautzberger notifications@github.com wrote:

> I think this test reveals a regular MathJax bug with empty annotation tags.
> 
> —

The test is wrong.

annotation and annotation-xml are both optional inside semantics and you
can have as many as you want in either order, but there should be an
initial element (which is not annotation or annotation-xml) which is the
thing being annotated so the last two expressions in that test

<semantics></semantics>

and

<semantics><annotation></annotation>/semantics>

are both in error but the test shows one as an error and one not, which I
don't think is justified.
(although rendering both as <mrow/> rather than an error is probably
reasonable

**Comment by pkra** [comment]
Thanks for the correction, @davidcarlisle. Either way, it's not really something ctop.js needs to concern itself with, wouldn't you say?

**Comment by christianp** [comment]
Yep. I've just made it so the code doesn't completely fail when a child node is missing.

**Comment by davidcarlisle** [comment]
On 2 May 2014 12:48, Christian Perfect notifications@github.com wrote:

> Yep. I've just made it so the code doesn't completely fail when a child
> node is missing.
> 
> —

sounds like a good plan:-)
